### PR TITLE
feat: add GTM setup

### PR DESCRIPTION
**Pull request by mattwendzina**
### 🧹 Please check off the following before submitting:

- [ ] Close issues
- [X] Request reviewers
- [X] Assign yourself
- [X] Add labels (e.g. `build:osc-ecommerce`)
- [ ] Add project
- [ ] Set project status
- [ ] Add milestone

---

## ☑ List all github issues that this PR will close:

N/A

## 📝 Description

Adds the Google Tag and some basic events

## ⛳️ Current behaviour (updates)

No GTM setup

## 🚀 New behaviour
- Loads Google Tag into the root which should propagate across all pages
- Sets up and sends PageView events

## 💣 Is this a breaking change (Yes/No):

No

## 🧾 Final Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible cross-device impacts (iOS, Android)
- [ ] I've considered possible cross-browser impacts (Chrome, Safari, Edge, Firefox)
- [ ] I've added tests to cover my changes (Unit, E2E, Integration, Contract, Functional, Acceptance etc)
- [ ] I've added or updated any documentation.

### Summary by CodeRabbit

- New Feature: Implemented Google Analytics tracking for page views and events. This will provide valuable insights into user behavior on the website.
- Refactor: Replaced the old Google Tag Manager scripts with new ones for improved tracking and performance.
- Chore: Added utility functions for interacting with Google Analytics, improving code modularity and maintainability.

### CODE REVIEW

File: packages/osc-recruitment/app/root.tsx [hunk 1]
```diff
@@ -145,32 +155,32 @@ export default function App() {
                 <Meta />
                 <Links />
                 <DynamicLinks />
-                {/* Google Tag Manager */}
```
[the remark this answers is not in the document]
Comment (mattwendzina) [reply packages/osc-recruitment/app/root.tsx:148]: Not sure what the NoScript tag is for tbh, but can move it back into their if that's better.

[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-recruitment/app/root.tsx:148]: Fair enough, was just curious. 
A `noscript` tag is something that will run if js is disabled for whatever reason, older sites sometimes use it to display a message "this site functions better with javascript" or whatever. 

From my memory when setting up gtm the guide tells you to put both in, but I've only ever done it on wordpress or php sites, so might be different with ones that use csr

**Review by stevenbister — APPROVED** [review]
As long as it's firing LGMT!

**Review by hi-im-si — CHANGES_REQUESTED** [review]
I am getting issues:
image

React dev tools is also saying:
image

might be something unrelated but getting a ton of hydration issues in the console too:
image

**Comment by mattwendzina** [comment]
> I am getting issues: image
> 
> React dev tools is also saying: image
> 
> might be something unrelated but getting a ton of hydration issues in the console too: image

Don't know why it's trying to look at localhost:3000, it should be localhost:3001 for recruitment 🤔 
I've added some instructions on the recruitment channel on how to test this, and in terms of your second comment this will be because it's coded to not run the scripts in development so you'll need to remove that check. 

That hydration issue has been there for a long time, we talked about it a few weeks ago in the office when Josh was working on some other problem. We also mentioned that there is the same issue with Academic Hub. Can't claim to know what the issue is here and whether it's anything to do with version clashes/being on the monorepo 🤷 I think Steve will probably be better placed to debug this and I'm assuming the fix will be needed/will sort both AH and recruitment.

**Comment by stevenbister** [comment]
Working for me!
image

**Review by hi-im-si — APPROVED** [review]

**Pull request merged**